CONTRIBUTING.md

```diff
@@ -1,34 +1,54 @@
+
+Ensure this repo is kept clean and linear repository history by using rebase and squash strategies effectively.
+
+
+
+
+
+
+# Pull Request Guidelines
+
+- **Initiate with Draft:** Begin every pull request in the `Draft` state to allow for incremental changes and feedback.
+- **Atomic Commits:** Ensure each commit is self-contained and represents a single logical change.
+- **Frequent Commits:** It's encouraged to commit and push changes to your branch frequently. This helps in keeping track of progress and facilitates easier collaboration.
+- **Rebase and Squash:** Before moving a PR out of draft, rebase your branch onto the latest `dev` branch and squash your commits to streamline the history.
+- **Logical Multiple Commits:** If multiple commits in a PR make logical sense, they can be kept, but strive to squash them into as few commits as possible to maintain clarity during reviews.
+
+- **Code Compliance:** Ensure your code adheres to the project’s formatting and linting guidelines.
+- **Testing:** Include test coverage and ensure your branch passes all required tests and CI/CD pipelines.
+- **Regular Updates:** Regularly update your branch with the latest changes from the `dev` branch using `--rebase` to avoid conflicts.
+- **Use Rebase, Avoid Merge:** Always use rebase instead of merge to integrate changes from `dev` to keep the history clean.
+- **No Direct Submissions to Main:** Direct submissions to the `main` branch are prohibited. All changes must go through the `dev` branch first.
+- **Review and Undraft:** Once your PR is complete, undraft it and request reviews to proceed with merging.
+
+## Standardized Formatting Guidelines
+
+To ensure consistency and maintainability of the codebase, all contributors are required to adhere to the following formatting guidelines:
+
+1. **Prettier Configuration:** Use the Google TypeScript Prettier settings. This helps in maintaining a uniform style for all TypeScript files.
+2. **Avoid Overusing Prettier Ignores:** While `/*prettier-ignore*/` can be used to bypass formatting for specific lines, it should be used sparingly. Excessive use can lead to inconsistent code styles across the project.
+3. **Linting:** Always run the project's linter before submitting a pull request. Ensure that your code complies with all linting rules set in the project configuration.
+4. **Code Readability:** Write clear and readable code. Use meaningful variable and function names and include comments where necessary to describe the purpose of the code blocks.
+5. **Follow Project Conventions:** Adhere to any additional formatting and coding standards that are documented in the project repository.
+6. **Commit Messages:** Follow a conventional commit message format for clarity and consistency, though it's not mandatory. [Commit Convention Guide](https://www.conventionalcommits.org/en/v1.0.0/)
+
+
+# Additional Guidelines
```

Comment:
We should remove the "Additional Guidelines" section, as the information it contains is already covered earlier in the document. Instead, let's add a new section called "Questions or Problems."

Here’s the suggested text for the new section:

## Questions or Problems?

If you have any questions or encounter any problems while contributing, don't hesitate to reach out for help. You can join the [discussions](https://github.com/shardeum/shardeum/discussions) on GitHub or join our community chat on [Discord](https://discord.com/invite/shardeum).